warn/scrapers/la.py

```diff
@@ -0,0 +1,394 @@
+import logging
+import os
+import re
+from datetime import datetime
+from pathlib import Path
+
+import pdfplumber
+from bs4 import BeautifulSoup
+
+from .. import utils
+from ..cache import Cache
+
+__authors__ = ["chriszs"]
+__tags__ = ["html", "pdf"]
+
+logger = logging.getLogger(__name__)
+
+
+def scrape(
+    data_dir: Path = utils.WARN_DATA_DIR,
+    cache_dir: Path = utils.WARN_CACHE_DIR,
+) -> Path:
+    """
+    Scrape data from Lousiana.
+
+    Keyword arguments:
+    data_dir -- the Path were the result will be saved (default WARN_DATA_DIR)
+    cache_dir -- the Path where results can be cached (default WARN_CACHE_DIR)
+
+    Returns: the Path where the file is written
+    """
+    # Fire up the cache
+    cache = Cache(cache_dir)
+
+    # The basic configuration for the scrape
+    state_code = "la"
+    base_url = "https://www.laworks.net/"
+    file_base = "Downloads_WFD"
+
+    # Download the root page
+    url = f"{base_url}Downloads/{file_base}.asp"
+    html = utils.get_url(url).text
+
+    # Save it to the cache
+    cache_key = f"{state_code}/{file_base}.html"
+    cache.write(cache_key, html)
+
+    # Parse out the links to WARN notice PDFs
+    links = _parse_links(html)
+
+    output_rows = []
+
+    for link in links:
+        if "WARN Notices" in link.text:
+            # Download the PDF
+            pdf_url = f"{base_url}{link['href']}"
+            pdf_path = _read_or_download(cache, state_code, pdf_url)
+
+            # Process the PDF
+            rows = _process_pdf(pdf_path)
+            output_rows.extend(rows)
+
+    # Insert a header row with clean column names.
+    # We are here assuming that the columns don't change between years
+    # and that one that contains "Employees Affected" will be clean.
+    output_rows = [list(filter(_is_clean_header, output_rows))[0]] + list(
+        filter(_is_not_header, output_rows)
+    )
```

Comment:
Picky thing. Can we name the first `output_rows` something else? Strictly speaking, I think it starts out as more than what we are going to output, and then gets filtered down at the tail end here. On my first read through the code it tripped me up a little.

Reply:
Sure, done.